File: sentimental-mario-less/mario.py
# print reversed half of pyramid

from cs50 import get_int


def main():
    while True:
        height = get_int("Height: ")
        if height > 0 and height < 9:
            break

    for i in range(height):
        print_row(height - i, i + 1)
        print()


def print_row(space, brick):
    for _ in range(space - 1, 0, -1):
        print(" ", end="")
    for _ in range(0, brick, 1):
        print("#", end="")


main()

File: mario-less/mario.c
// Build reversed pyramid based on input 

#include <cs50.h>
#include <stdio.h>

void print_row(int space, int brick);

int main(void)
{
    // ask for height
    int n;
    do
    {
        n = get_int("Height: ");
    }
    while (n < 1);

    // print pyramid of that height
    for (int i = 0; i < n; i++)
    {
        // print row of bricks
        print_row(n - i, i + 1);
    }
}

void print_row(int space, int brick)
{
    // print spaces
    for (int i = space - 1; i > 0; i--)
    {
        printf(" ");
    }
    // print bricks
    for (int i = 0; i < brick; i++)
    {
        printf("#");
    }
    printf("\n");
}
